### PR TITLE
Feature/reconcile cluster using update

### PR DESCRIPTION
**Pull request by Tatsinnit**
This PR is draft and it enables - Reconcile cluster operation which is an empty update on the cluster.

### CODE REVIEW

File: src/commands/aksReconcileCluster/aksReconcileCluster.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  const cluster = getAksClusterTreeItem(target, cloudExplorer);
+  if (failed(cluster)) {
+    vscode.window.showErrorMessage(cluster.error);
+    return;
+  }
+
+  const clusterName = cluster.result.name;
+
+  const clusterProvisioingState = await determineProvisioningState(cluster.result, clusterName);
```
Comment (peterbom) [inline src/commands/aksReconcileCluster/aksReconcileCluster.ts:22]: Looks like this is unused and could be removed...unless we can figure out the circumstances under which reconciliation is appropriate, and this is relevant?

File: src/commands/aksReconcileCluster/aksReconcileCluster.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { reconcileUsingUpdateInCluster, determineProvisioningState, getAksClusterTreeItem } from '../utils/clusters';
```
Comment (hsubramanianaks) [inline src/commands/aksReconcileCluster/aksReconcileCluster.ts:4]: I think determineProvisioningState needs to be removed from the imports as well.

**Comment by sprab** [comment]
Tested this with multiple scenarios and works as expected.

**Review by sprab — APPROVED** [review]
Tested with multiple scenarios and works as expected.

**Review by sprab — COMMENTED** [review]
Tested this with various scenarios that includes changes made to MC resources outside of AKS RP and the reconcile works as expected.

File: package.json [hunk 3]
```diff
@@ -232,6 +232,10 @@
                 "command": "aks.aksAbortLastOperationInCluster",
                 "title": "Abort Last Operation"
             },
+            {
+                "command": "aks.aksReconcileCluster",
+                "title": "Reconcile Operation"
```
Comment (peterbom) [inline package.json:237]: Could we call this just 'Reconcile', or maybe 'Reconcile Cluster'?

As it is, it sounds a bit like 'reconcile the operation', which doesn't make sense.

**Review by hsubramanianaks — APPROVED** [review]
@Tatsinnit  looks good, tested with my own clusters and it works fine. Thanks for this feature, it will solve multiple user AKS cluster problems.

**Review by peterbom — APPROVED** [review]
Looks good - thanks @Tatsinnit! I'd like to do some more work in future around when it's appropriate to perform this action, and guide users accordingly, but that'll require a lot more analysis. For now, let's make this functionality available.

**Pull request merged**